test/e2e/kubeproxy.go

```diff
@@ -417,13 +423,32 @@ func (config *KubeProxyTestConfig) setup() {
 		selectorName: "true",
 	}
 
-	By("Getting ssh-able hosts")
-	hosts, err := NodeSSHHosts(config.f.Client)
-	Expect(err).NotTo(HaveOccurred())
-	if len(hosts) == 0 {
-		Failf("No ssh-able nodes")
+	config.nodes = []string{}
+	if providerIs(providersWithSSH...) {
+		By("Getting ssh-able hosts")
+		hosts, err := NodeSSHHosts(config.f.Client)
+		Expect(err).NotTo(HaveOccurred())
+		for _, h := range hosts {
+			config.nodes = append(config.nodes, strings.TrimSuffix(h, ":22"))
+		}
+	} else {
+		// fall back to legacy IP
```

Comment:
It does not make sense in general I think, only for this test. But I removed this now, following your comment above to skip the whole test. I agree a net=host pod is the way to go.